experimental/bot/logger/telemetrylogger/telemetry_logger.go

```diff
@@ -0,0 +1,74 @@
+package telemetrylogger
+
+import (
+	"fmt"
+
+	"github.com/mattermost/mattermost-plugin-api/experimental/bot/logger"
+	"github.com/mattermost/mattermost-plugin-api/experimental/telemetry"
+)
+
+type telemetryLogger struct {
+	logger.Logger
+	logLevel logger.LogLevel
+	tracker  telemetry.Tracker
+}
+
+/*
+New promotes the provided logger into a telemetry logger, storing all events below the logLevel
+through the tracker.
+
+- l Logger: A logger to promote.
+
+- logLevel: The highest type of message to be stored in telemetry.
```

Comment:
What is the use case of a logeLevel? I can't think of one.

Reply:
We may want to send to telemetry only Error level logs, or anything that is not a Debug log. That is the reason to be of logLevel.

Reply:
I still don't quite get the use case. When is the decision to send on a certain level made? On build time? Is there a case where it's decided on run time?

Reply:
In this case, for telemetry logger, I cannot think any "real scenario" where deciding on run time make sense. For Admin logger, we may want to have a setting somewhere, and change the log level based on the setting.
But on the weird scenario that the plugin developer uses this telemetry logger to log into its own logging system, and has a setting to decide which logLevel to log, that would be a use case for this to be set on run-time.

Reply:
I'm 1/5 that we should go with the simpler approach here and drop the logLevel for the telemetry logger. And if we come up with a real world use case, we can still add it later.

Reply:
But what about the case we want to decide on build time? Like "log to telemetry only error messages"?

Reply:
If the decision is made on build time there are already tools like to build flags to handle that case.

Reply:
I will keep the logLevel for versatility, even if it makes since slightly more complex. Nevertheless, I keep you comments in mind.